### PR TITLE
checkstyle: prohibit API client repackaged Guava

### PR DESCRIPTION
**Pull request by dhalperi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Apparently the IllegalImport check only blocks packages,
so we had to move to Regexp to get individual classes.

As a bonus, this enforcement let us remove two bogus dependencies.
Smaller JARs for the win!

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @kennknowles

File: runners/spark/pom.xml [hunk 1]
```diff
@@ -155,10 +155,6 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>com.google.http-client</groupId>
```
Comment (bjchambers) [inline runners/spark/pom.xml]: This seems unrelated to PR description.

File: sdks/java/build-tools/src/main/resources/beam/checkstyle.xml [hunk 2]
```diff
@@ -100,7 +100,11 @@ page at http://checkstyle.sourceforge.net/config.html -->
     </module>
 
     <module name="IllegalImport">
-      <property name="illegalPkgs" value="com.google.api.client.repackaged, com.google.api.client.util.util.ByteStreams, com.google.api.client.util.Charsets, com.google.api.client.util.Collections2, com.google.api.client.util.Joiner, com.google.api.client.util.Lists, com.google.api.client.util.Maps, com.google.api.client.util.Objects, com.google.api.client.util.Preconditions, com.google.api.client.util.Sets, com.google.api.client.util.Strings, com.google.api.client.util.Throwables, io.netty.util.internal"/>
+      <property name="illegalPkgs" value="com.google.api.client.repackaged, io.netty.util.internal"/>
+    </module>
+
+    <module name="RegexpSinglelineJava">
```
Comment (bjchambers) [inline sdks/java/build-tools/src/main/resources/beam/checkstyle.xml]: Maybe a comment about why we're using Regexp rather than IllegalImport? My guess is that illegalPkgs doesn't work with classes.

**Comment by bjchambers** [comment]
Minor nits then LGTM
